lib/javascript/JavascriptParser.js

```diff
@@ -417,23 +421,42 @@ class JavascriptParser extends Parser {
 
 				const left = this.evaluateExpression(expr.left);
 				if (!left) return;
+				let returnRight;
+				/** @type {boolean|null} */
+				let allowedRight;
 				if (expr.operator === "&&") {
 					const leftAsBool = left.asBool();
 					if (leftAsBool === false) return left.setRange(expr.range);
-					if (leftAsBool !== true) return;
+					returnRight = leftAsBool === true;
+					allowedRight = false;
 				} else if (expr.operator === "||") {
 					const leftAsBool = left.asBool();
 					if (leftAsBool === true) return left.setRange(expr.range);
-					if (leftAsBool !== false) return;
+					returnRight = leftAsBool === false;
+					allowedRight = true;
 				} else if (expr.operator === "??") {
 					const leftAsNullish = left.asNullish();
 					if (leftAsNullish === false) return left.setRange(expr.range);
-					if (leftAsNullish !== true) return;
+					returnRight = leftAsNullish === true;
```

Comment:
```suggestion
					if (leftAsNullish !== true) return;
					returnRight = true;
```

This can early return here to skip evaluating the right side.